### PR TITLE
 [Feature request] Compile with database output 

### PR DESCRIPTION
**Pull request by spothound**
Hello team

This Pr aims to add an option for building a package with database output support (MySQL and pgSQL).

Closes https://github.com/wazuh/wazuh-packages/issues/437

I've added some required dependencies to all deb and rpm Dockerfiles and modified the SPECS and building script to add this option.

### CODE REVIEW

**Comment by grg121** [comment]
# Testing in Jenkins

Build all rpms (all architectures) https://jenkins-production.wazuh.info/view/Packages/job/Packages_builder_rpm/4917/console
Build all debs (all architectures) https://jenkins-production.wazuh.info/view/Packages/job/Packages_builder_deb/4020/console 

(These build without --database flag, just a regular build to check that nothing is broken)

**Comment by grg121** [comment]
New tests:

- https://jenkins-production.wazuh.info/view/Packages/job/Packages_builder_deb/4050/console
- https://jenkins-production.wazuh.info/view/Packages/job/Packages_builder_rpm/4948/console

File: debs/generate_debian_package.sh [hunk 1]
```diff
@@ -147,6 +148,7 @@ help() {
     echo "    -s, --store <path>        [Optional] Set the destination path of package. By default, an output folder will be created."
     echo "    -p, --path <path>         [Optional] Installation path for the package. By default: /var/ossec."
     echo "    -d, --debug               [Optional] Build the binaries with debug symbols. By default: no."
+    echo "    --database <mysql/pgsql>  [Optional] Build with database output support for mysql or pgsql."
```
Comment (BraulioV) [inline debs/generate_debian_package.sh:151]:
```suggestion
    echo "    --database <mysql/pgsql>  [Optional] Build with database output support for mysql or pgsql. This option is disabled by default."
```

File: debs/generate_debian_package.sh [hunk 2]
```diff
@@ -74,7 +75,7 @@ build_deb() {
         ${CUSTOM_CODE_VOL} \
         ${CONTAINER_NAME} ${TARGET} ${BRANCH} ${ARCHITECTURE} \
         ${REVISION} ${JOBS} ${INSTALLATION_PATH} ${DEBUG} \
-        ${CHECKSUM} ${PACKAGES_BRANCH} ${USE_LOCAL_SPECS} ${LOCAL_SOURCE_CODE} || return 1
+        ${CHECKSUM} ${PACKAGES_BRANCH} ${USE_LOCAL_SPECS} "${LOCAL_SOURCE_CODE}" "DATABASE=${DATABASE_OUTPUT}" || return 1
```
Comment (BraulioV) [inline debs/generate_debian_package.sh:78]:
```shell
...
DATABASE_FLAG="DATABASE=${DATABASE_OUTPUT}"
...

        ${CHECKSUM} ${PACKAGES_BRANCH} ${USE_LOCAL_SPECS} \ 
        ${LOCAL_SOURCE_CODE} ${DATABASE_FLAG} || return 1
```
To keep the consistency with other configuration variables, you need to set a variable for the `DATABASE` flag and then use it in the `docker run` line. Also, break the line to make easier reading the command.

File: rpms/generate_rpm_package.sh [hunk 3]
```diff
@@ -93,7 +94,7 @@ build_rpm() {
         ${CONTAINER_NAME} ${TARGET} ${BRANCH} ${ARCHITECTURE} \
         ${JOBS} ${REVISION} ${INSTALLATION_PATH} ${DEBUG} \
         ${CHECKSUM} ${PACKAGES_BRANCH} ${USE_LOCAL_SPECS} ${SRC} \
-        ${LEGACY} ${LOCAL_SOURCE_CODE} || return 1
+        ${LEGACY} "${LOCAL_SOURCE_CODE}" "${DATABASE_OUTPUT}" || return 1
```
Comment (BraulioV) [inline rpms/generate_rpm_package.sh:97]: As you did here. Check the last comment in the Debian script and name the variables equally in both scripts. 

PS: you don't need to quote the variables if they don't contain any spaces.

**Comment by grg121** [comment]
New tests:

    https://jenkins-production.wazuh.info/view/Packages/job/Packages_builder_deb/4075/console
    https://jenkins-production.wazuh.info/view/Packages/job/Packages_builder_rpm/4973/console

@grg121

File: debs/Debian/amd64/build.sh [hunk 4]
```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
```
Comment (BraulioV) [inline debs/Debian/amd64/build.sh:1]: Remove this file

File: debs/build.sh [hunk 5]
```diff
@@ -22,18 +22,19 @@ checksum=$8
...
 local_source_code=${11}
+database_output=${12}
 
 if [ -z "${package_release}" ]; then
     package_release="1"
 fi
 
 if [ ${build_target} = "api" ]; then
-    if [ -z "${local_source_code}" ]; then
+    if [ "${local_source_code}" = 'No' ]; then
```
Comment (BraulioV) [inline debs/build.sh:32]:
```suggestion
    if [ "${local_source_code}" = 'no' ]; then
```

File: debs/build.sh [hunk 6]
```diff
@@ -22,18 +22,19 @@ checksum=$8
...
         curl -sL https://github.com/wazuh/wazuh-api/tarball/${wazuh_branch} | tar zx
     fi
     wazuh_version="$(grep version wazuh*/package.json | cut -d '"' -f 4)"
 else
-    if [ -z "${local_source_code}" ]; then
+    if [ "${local_source_code}" = 'No' ]; then
```
Comment (BraulioV) [inline debs/build.sh:37]:
```suggestion
    if [ "${local_source_code}" = 'no' ]; then
```

File: debs/build.sh [hunk 7]
```diff
@@ -63,6 +64,7 @@ cd ${build_dir}/${build_target} && tar -czf ${package_full_name}.orig.tar.gz "${
 
 # Configure the package with the different parameters
 sed -i "s:RELEASE:${package_release}:g" ${sources_dir}/debian/changelog
+sed -i "s:export DATABASE_OUTPUT=.*:export DATABASE_OUTPUT=DATABASE=${database_output}:g" ${sources_dir}/debian/rules
```
Comment (BraulioV) [inline debs/build.sh:67]:
```suggestion
sed -i "s:export DATABASE_OUTPUT=.*:export DATABASE_OUTPUT=\"DATABASE='${database_output}'\":g" ${sources_dir}/debian/rules
```

File: debs/generate_debian_package.sh [hunk 8]
```diff
@@ -33,7 +33,8 @@ CHECKSUM="no"
 PACKAGES_BRANCH="master"
 USE_LOCAL_SPECS="no"
 LOCAL_SPECS="${CURRENT_PATH}"
-LOCAL_SOURCE_CODE=""
+LOCAL_SOURCE_CODE="No"
```
Comment (BraulioV) [inline debs/generate_debian_package.sh:36]:
```suggestion
LOCAL_SOURCE_CODE="no"
```

File: debs/generate_debian_package.sh [hunk 9]
```diff
@@ -58,7 +59,7 @@ build_deb() {
     cp build.sh ${DOCKERFILE_PATH}
 
     # Create an optional parameter to share the local source code as a volume
-    if [ ! -z "${LOCAL_SOURCE_CODE}" ]; then
+    if [ "${LOCAL_SOURCE_CODE}" != 'No' ]; then
```
Comment (BraulioV) [inline debs/generate_debian_package.sh:62]:
```suggestion
    if [ "${LOCAL_SOURCE_CODE}" != 'no' ]; then
```

File: rpms/CentOS/7/ppc64le/Dockerfile [hunk 10]
```diff
@@ -2,14 +2,14 @@ FROM ppc64le/centos:7
...
-RUN yum -y install centos-release-scl 
+RUN yum -y install centos-release-scl
 RUN mv /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo /etc/yum.repos.d/CentOS-SCLo-scl-rh.repo.old
 RUN mv /etc/yum.repos.d/CentOS-SCLo-scl.repo /etc/yum.repos.d/CentOS-SCLo-scl.repo.old
 
 
 RUN yum -y install gcc make git \
     openssh-clients rpm-build sudo gnupg \
-    automake autoconf libtool policycoreutils-python \
+    automake autoconf libtool policycoreutils-python mysql-devel postgresql-devel \
```
Comment (BraulioV) [inline rpms/CentOS/7/ppc64le/Dockerfile:12]:
```suggestion
    automake autoconf libtool policycoreutils-python \
    mysql-devel postgresql-devel \
```

File: rpms/build.sh [hunk 11]
```diff
@@ -37,12 +38,12 @@ if [ "${debug}" = "no" ]; then
 fi
 
 if [ ${build_target} = "api" ]; then
-    if [ -z "${local_source_code}" ]; then
+    if [ "${local_source_code}" = 'No' ]; then
```
Comment (BraulioV) [inline rpms/build.sh:41]:
```suggestion
    if [ "${local_source_code}" = 'no' ]; then
```

File: rpms/build.sh [hunk 12]
```diff
@@ -37,12 +38,12 @@ if [ "${debug}" = "no" ]; then
...
         curl -sL https://github.com/wazuh/wazuh-api/tarball/${wazuh_branch} | tar zx
     fi
     wazuh_version="$(grep version wazuh*/package.json | cut -d '"' -f 4)"
 else
-    if [ -z "${local_source_code}" ]; then
+    if [ "${local_source_code}" = 'No' ]; then
```
Comment (BraulioV) [inline rpms/build.sh:46]:
```suggestion
    if [ "${local_source_code}" = 'no' ]; then
```

File: rpms/generate_rpm_package.sh [hunk 13]
```diff
@@ -40,7 +40,8 @@ PACKAGES_BRANCH="master"
 CHECKSUMDIR=""
 CHECKSUM="no"
 USE_LOCAL_SPECS="no"
-LOCAL_SOURCE_CODE=""
+LOCAL_SOURCE_CODE="No"
```
Comment (BraulioV) [inline rpms/generate_rpm_package.sh:43]:
```suggestion
LOCAL_SOURCE_CODE="no"
```

File: rpms/generate_rpm_package.sh [hunk 14]
```diff
@@ -76,7 +77,7 @@ build_rpm() {
     fi
 
     # Create an optional parameter to share the local source code as a volume
-    if [ ! -z "${LOCAL_SOURCE_CODE}" ]; then
+    if [ "${LOCAL_SOURCE_CODE}" != 'No' ]; then
```
Comment (BraulioV) [inline rpms/generate_rpm_package.sh:80]:
```suggestion
    if [ "${LOCAL_SOURCE_CODE}" != 'no' ]; then
```

File: rpms/generate_rpm_package.sh (hunk 13 above)
Comment (grg121) [reply rpms/generate_rpm_package.sh]: https://github.com/wazuh/wazuh-packages/pull/446/commits/002e5300c324caf125c27a9fc85852a6c19e0945

File: rpms/build.sh (hunk 12 above)
Comment (grg121) [reply rpms/build.sh]: https://github.com/wazuh/wazuh-packages/pull/446/commits/002e5300c324caf125c27a9fc85852a6c19e0945

File: rpms/build.sh (hunk 11 above)
Comment (grg121) [reply rpms/build.sh]: https://github.com/wazuh/wazuh-packages/pull/446/commits/002e5300c324caf125c27a9fc85852a6c19e0945

File: debs/generate_debian_package.sh (hunk 9 above)
Comment (grg121) [reply debs/generate_debian_package.sh]: https://github.com/wazuh/wazuh-packages/pull/446/commits/002e5300c324caf125c27a9fc85852a6c19e0945

File: debs/generate_debian_package.sh (hunk 8 above)
Comment (grg121) [reply debs/generate_debian_package.sh]: https://github.com/wazuh/wazuh-packages/pull/446/commits/002e5300c324caf125c27a9fc85852a6c19e0945

File: debs/build.sh (hunk 6 above)
Comment (grg121) [reply debs/build.sh]: https://github.com/wazuh/wazuh-packages/pull/446/commits/002e5300c324caf125c27a9fc85852a6c19e0945

File: debs/build.sh (hunk 5 above)
Comment (grg121) [reply debs/build.sh]: https://github.com/wazuh/wazuh-packages/pull/446/commits/002e5300c324caf125c27a9fc85852a6c19e0945

File: debs/Debian/amd64/build.sh (hunk 4 above)
Comment (grg121) [reply debs/Debian/amd64/build.sh]: https://github.com/wazuh/wazuh-packages/pull/446/commits/f32afb3cfc9b039a7ea315e39874efab1fdcffab

**Comment by grg121** [comment]
https://jenkins-production.wazuh.info/view/Packages/job/Packages_builder_deb/4103/console
https://jenkins-production.wazuh.info/view/Packages/job/Packages_builder_rpm/4999/console

**Pull request closed without merging**